controllers/trivy_vuln_crds_controller.go

```diff
@@ -0,0 +1,219 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"github.com/aquasecurity/trivy-operator/pkg/apis/aquasecurity/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/json"
+	"os"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strconv"
+)
+
+const LagoonProjectLabel = "lagoon.sh/project"
+const TrivyVulnReportNamespace = "trivy-operator.resource.namespace"
+
+type TrivyVulnerabilityReportReconciler struct {
+	Client         client.Client
+	Scheme         *runtime.Scheme
+	MessageQWriter func(data []byte) error
+	WriteToQueue   bool
+	PredicateData  interface{}
+}
+
+type ProblemsPayload struct {
+	Problems []interface{} `json:"problems"`
+}
+
+// predicate that filters out non-lagoon projects
+func (r *TrivyVulnerabilityReportReconciler) lagoonProjectLabelsOnlyPredicate() predicate.Predicate {
+
+	return predicate.Funcs{
+		CreateFunc: func(event event.CreateEvent) bool {
+			if r.isLagoonProject(event.Object) &&
+				!insightsProcessedAnnotationExists(event.Object) {
+				return true
+			}
+			return false
+		},
+		UpdateFunc: func(event event.UpdateEvent) bool {
+			if r.isLagoonProject(event.ObjectNew) &&
+				!insightsProcessedAnnotationExists(event.ObjectNew) {
+				return true
+			}
+			return false
+		},
+		DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
+			return false
+		},
+	}
+}
+
+func (r *TrivyVulnerabilityReportReconciler) isLagoonProject(event client.Object) bool {
+	var namespace string
+	for k, v := range event.GetLabels() {
+		if k == TrivyVulnReportNamespace {
+			namespace = v
+		}
+	}
+
+	nsObj := &corev1.Namespace{}
+	nsObj.SetName(namespace)
+
+	// retrieve the ns object from the k8 API
+	err := r.Client.Get(context.Background(), client.ObjectKeyFromObject(nsObj), nsObj)
+	if err != nil {
+		fmt.Printf("Failed to get Namespace %q: %v\n", namespace, err)
+		os.Exit(1)
+	}
+
+	// check the ns if this is a lagoon project
+	isLagoonProject := false
+	for k, v := range nsObj.Labels {
```

Comment:
if nsObj.Labels is a map, you could do this
```
val, ok := myMap["foo"]
// If the key exists
if ok {
    // Do something
}
```
rather than iterate over all of em.